test/integration-new/cni/cni_version_test.go

```diff
@@ -0,0 +1,104 @@
+package cni
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/aws/services"
+	"github.com/pkg/errors"
+
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	describeAddonOutput *eks.DescribeAddonOutput
+	err                 error
+	initialCNIVersion   string
+	finalCNIVersion     string
+)
+
+var _ = Describe("test cluster upgrade/downgrade", func() {
+
+	It("should apply initial addon version successfully", func() {
+		By("getting initial cni version")
+		if len(f.Options.InitialCNIVersion) == 0 {
+			err = errors.Errorf("%s must be set!", "initial-version")
+		}
+		Expect(err).ToNot(HaveOccurred())
+		initialCNIVersion = f.Options.InitialCNIVersion
+		ApplyAddOn(initialCNIVersion)
+
+	})
+
+	Context("when testing host networking on initial version", func() {
+		HostNetworkingTest()
+		PodTrafficTest()
+		ServiceConnectivityTest()
+	})
+
+	It("should apply final addon version successfully", func() {
+		By("getting final cni version")
+		if len(f.Options.FinalCNIVersion) == 0 {
+			err = errors.Errorf("%s must be set!", "final-version")
+		}
+		Expect(err).ToNot(HaveOccurred())
+		finalCNIVersion = f.Options.FinalCNIVersion
+		ApplyAddOn(finalCNIVersion)
+
+	})
+
+	Context("when testing host networking on final version", func() {
+		HostNetworkingTest()
+		PodTrafficTest()
+		ServiceConnectivityTest()
+	})
+
+})
+
+func ApplyAddOn(versionName string) {
+
+	By("getting the current addon")
+	describeAddonOutput, err = f.CloudServices.EKS().DescribeAddon("vpc-cni", f.Options.ClusterName)
+	if err == nil {
+		By("checking if the current addon is same as addon to be applied")
+		if *describeAddonOutput.Addon.AddonVersion != versionName {
+
+			By("deleting the current vpc cni addon ")
+			_, err = f.CloudServices.EKS().DeleteAddon("vpc-cni", f.Options.ClusterName)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = f.CloudServices.EKS().DescribeAddon("vpc-cni", f.Options.ClusterName)
+			for err == nil {
+				time.Sleep(5 * time.Second)
+				_, err = f.CloudServices.EKS().DescribeAddon("vpc-cni", f.Options.ClusterName)
+
+			}
+
+			By(fmt.Sprintf("By applying addon %s\n", versionName))
+			_, err = f.CloudServices.EKS().CreateAddon(services.AddOnInput{AddonName: "vpc-cni", ClusterName: f.Options.ClusterName, AddonVersion: versionName})
+			Expect(err).ToNot(HaveOccurred())
+
+		}
+	} else {
+		By(fmt.Sprintf("By applying addon %s\n", versionName))
+		_, err = f.CloudServices.EKS().CreateAddon(services.AddOnInput{AddonName: "vpc-cni", ClusterName: f.Options.ClusterName, AddonVersion: versionName})
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	var status = ""
+
+	By("waiting for  addon to be ACTIVE")
+	for status != "ACTIVE" {
```

Comment:
I think this can be extracted to utility function similar to daemonset.go. So this logic can be extended.